lib/flight_job/models/job.rb

```diff
@@ -102,6 +107,20 @@ def save
       end
     end
 
+    def pass_filter?(opts)
+      return true unless opts && (opts.id || opts.script || opts.state)
+      job_attributes.each_pair do |key, _|
+        if opts[key]
+          return false unless super(opts[key],job_attributes[key])
+        end
+      end
+      true
+    end
+
+    def job_attributes
+      @job_attributes ||= OpenStruct.new(id: id, script: script_id, state: self.decorate.state)
+    end
```

Comment:
I'm not sure that having `Matcher` defined as a module is the best way to do this.  The use of `super` halfway through `Job#pass_filter?` was a little confusing and its not clear what `Job#job_attributes` is for.

I would have expected `Job#pass_filter?` to be something more like


```ruby
def pass_filter?(opts)
  @_matcher ||= Matcher.new(self.decorate, opts, {id: :id, script: :script_id, state: :state})
  @_matcher.matches?
end
```

or perhaps

```ruby
def pass_filter?(filters)
  @_matcher ||= Matcher.new(filters, {id: id, script: script_id, state: decorate.state})
  @_matcher.pass_filter?
end
```

This much shorter `pass_filter?` method has `Job` declare what is to be filtered and how the option names match to the job values.  That way when we extend filtering to tasks, templates and/or scripts, very little code is required in `Task`, etc. as most all of the implementation is in `Matcher`.

Does that make sense?